### PR TITLE
feat: V4 create

### PR DESCRIPTION
**Pull request by jjtang1985**
## Context

Please provide a description of what your change does and why it is needed.

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
- [ ] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [ ] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

**Comment by FrankEssenberger** [comment]
In this case the context is rather clear: create for odata v4. But in general one or two sentences would be nice.

File: packages/core/src/odata/v4/request-builder/create-request-builder.ts [hunk 1]
```diff
@@ -86,7 +87,7 @@ export class CreateRequestBuilder<EntityT extends Entity>
       .then(request => request.execute())
       .then(response =>
         deserializeEntity(
-          response.data.d,
+          extractData(response),
```
Comment (FrankEssenberger) [inline packages/core/src/odata/v4/request-builder/create-request-builder.ts:90]: could you make a second PR, where you look for the many hard coded `.data.d` usage and replace them all.

Additional optimization: Is this something we can know from the system we are calling for example S/4 use `data.d`, C/4 something I do not know, concur XYZ. Perhaps it makes then sense to include this extraction in the builder or so. I do not know.

File: packages/core/src/odata/v4/request-builder/create-request-builder.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
 
-import { errorWithCause } from '@sap-cloud-sdk/util';
+import { errorWithCause, MapType } from '@sap-cloud-sdk/util';
```
Comment (FrankEssenberger) [inline packages/core/src/odata/v4/request-builder/create-request-builder.ts:3]: I compared this once with the v2 create request builder. I did the same also for the get all and get by key and they are also very similar. Please create a follow up ticket that we do not forget to reduce this code duplication. Obvious way would be base classes but mixin could be a more TS way to do that.

File: packages/core/test/test-util/testPostRequestOutcome.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
```
Comment (FrankEssenberger) [inline packages/core/test/test-util/testPostRequestOutcome.ts:1]: the file name is not kebap case

File: test-packages/integration-tests/test/test-data/single-test-entity-response-v4.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
```
Comment (FrankEssenberger) [inline test-packages/integration-tests/test/test-data/single-test-entity-response-v4.ts:1]: Also create a ticket that we remember to setup a real E2E test with a CAP service or one of the two v4 services.

File: packages/core/src/odata/v4/request-builder/create-request-builder.ts (hunk 1 above)
Comment (jjtang1985) [reply packages/core/src/odata/v4/request-builder/create-request-builder.ts:90]: - had a discussion with Marika and decided to make abstract request builders and move the `.data.d` to a single file -> [ticket](https://github.wdf.sap.corp/MA/scp-rocket/issues/1257).
- for different systems, if there is a clear rule, we can use a system-to-extractData mapping. However, even for c/4, the known rule is only for v2 not v4. Also, we decided to also support collection in v2, so SFSF can also make use of the collection type now.

File: packages/core/src/odata/v4/request-builder/create-request-builder.ts (hunk 2 above)
Comment (jjtang1985) [reply packages/core/src/odata/v4/request-builder/create-request-builder.ts:3]: see comment above.

File: test-packages/integration-tests/test/test-data/single-test-entity-response-v4.ts (hunk 4 above)
Comment (jjtang1985) [reply test-packages/integration-tests/test/test-data/single-test-entity-response-v4.ts:1]: It's on the radar, but you are right, a ticket is good for tracking.
-> https://github.wdf.sap.corp/MA/scp-rocket/issues/1258

**Pull request merged**